test/mocks/router/mocks.h

```diff
@@ -182,6 +182,9 @@ class MockRateLimitPolicy : public RateLimitPolicy {
 
 class TestShadowPolicy : public ShadowPolicy {
 public:
+  TestShadowPolicy(const std::string& cluster = "", const std::string& runtime_key = "",
```

Comment:
use string_view here instead

doesnt matter a whole lot since this is in test you're doing empty strings (so SSO applies), but this ends up instantiating new string objects every time any of the default values are used

Reply:
✅